Sources/BraveWallet/Crypto/NFTImageView.swift

```diff
@@ -20,20 +20,26 @@ struct NFTImageView<Placeholder: View>: View {
   }
   
   var body: some View {
-    if urlString.hasPrefix("data:image/") {
-      WebImageReader(url: URL(string: urlString)) { image, isFinished in
-        if let image = image {
-          Image(uiImage: image)
+    if let url = URL(string: urlString) {
+      if url.absoluteString.hasPrefix("data:image/") {
+        WebImageReader(url: url) { image, _ in
+          if let image = image {
+            Image(uiImage: image)
+              .resizable()
+              .aspectRatio(contentMode: .fit)
+          } else {
+            placeholder()
+          }
+        }
+      } else if url.isSecureWebPage() {
+        if url.absoluteString.hasSuffix(".gif") {
+          WebImage(url: url)
```

Comment:
Not sure that this is technically enough to determine if its a gif, but maybe fine for the time being? Does BraveCore's implementation not download gif's correctly? Or is it just about display?

Reply:
There is another field in the metadata `animation_url`, however, it's a html (technically, this field is for multi-media support. It could a gif, a video or a piece of audio) I guess WKWebView all the way 🥲 
The BraveCore's implementation is able to download one frame of the gif which will be a static image.

Reply:
Okay sounds good we can keep this for now